test/builtInContext.js

```diff
@@ -0,0 +1,153 @@
+                                "longitude": "0.1278 W",
+                                "latitude": "51.5074 N"
+                            }
+                        }
+                    }
+                }
+            }
+        };
+
+        dummyInvalidContext = { "builtin": {"Dsfds": "Sdfs"}};
+        validPath = "currentConversation.lastReferencedLocation";
+        invalidPath = "currentConversation.hello.world";
+        dummyValue = {
+            "@type": "http://www.ibm.com/watson-assistant-solutions/ontology/PlaceReference",
+            "object": {
+                "@type": "http://schema.org/Place",
+                "name": "John Hancock Tower Boston",
+                "address": {
+                    "@type": "http://schema.org/PostalAddress",
+                    "streetAddress": "120 St James Ave",
+                    "addressLocality": "Boston",
+                    "postalCode": "MA 02116",
+                    "addressCountry": "USA"
+                },
+                "geo": {
+                    "@type": "http://schema.org/GeoCoordinates",
+                    "longitude": "no",
+                    "latitude": "where"
+                }
+            }
+        };
+        dummyInvalidValue = {
+            "object": {
+                "@type": "http://schema.org/Place",
+                "name": "John Hancock Tower Boston",
+                "address": {
+                    "@type": "http://schema.org/PostalAddress",
+                    "streetAddress": "120 St James Ave",
+                    "addressLocality": "Boston",
+                    "postalCode": "MA 02116",
+                    "addressCountry": "USA"
+                },
+                "geo": {
+                    "@type": "http://schema.org/GeoCoordinates",
+                    "longitude": "no",
+                    "latitude": "where"
+                }
+            }
+        };
+        done();
+    });
+
+    describe('Test set built-in context', function() {
+        it('Try setting invalid built-in context property path', function (done) {
+            const result = handler.setBuiltInContextProperty(dummyValidContext, invalidPath, dummyValue);
+            result.valid.should.equal(false);
+            result.status.should.equal(404);
+            done()
+        });
+        it('Try setting valid built-in context property path', function(done) {
+            const result = handler.setBuiltInContextProperty(dummyValidContext, validPath, dummyValue);
+            result.valid.should.equal(true);
+            JSON.stringify(result.obj.currentConversation.lastReferencedLocation).should.equal(JSON.stringify(dummyValue));
+            done();
+        });
+        it('Try setting invalid built-in context property value', function(done) {
+            const result = handler.setBuiltInContextProperty(dummyValidContext, validPath, dummyInvalidValue);
+            result.valid.should.equal(false);
```

Comment:
should check the defined behavior for 'obj' too. Make sure that it is what it is supposed to be when valid is false.